sensing/autoware_pointcloud_preprocessor/include/autoware/pointcloud_preprocessor/distortion_corrector/distortion_corrector.hpp

```diff
@@ -47,74 +47,95 @@
 namespace autoware::pointcloud_preprocessor
 {
 
-class DistortionCorrectorBase
+struct AngleConversion
 {
-public:
-  virtual bool pointcloud_transform_exists() = 0;
-  virtual bool pointcloud_transform_needed() = 0;
-  virtual std::deque<geometry_msgs::msg::TwistStamped> get_twist_queue() = 0;
-  virtual std::deque<geometry_msgs::msg::Vector3Stamped> get_angular_velocity_queue() = 0;
-
-  virtual void processTwistMessage(
-    const geometry_msgs::msg::TwistWithCovarianceStamped::ConstSharedPtr twist_msg) = 0;
-  virtual void processIMUMessage(
-    const std::string & base_frame, const sensor_msgs::msg::Imu::ConstSharedPtr imu_msg) = 0;
-  virtual void setPointCloudTransform(
-    const std::string & base_frame, const std::string & lidar_frame) = 0;
-  virtual void initialize() = 0;
-  virtual void undistortPointCloud(bool use_imu, sensor_msgs::msg::PointCloud2 & pointcloud) = 0;
+  // Equation for the conversion between sensor azimuth coordinates and Cartesian coordinates:
+  // sensor azimuth coordinates = offset_rad + sign * cartesian coordinates;
+  // offset_rad is restricted to be a multiple of 90, and sign is restricted to be 1 or -1.
+  float offset_rad{0};
+  float sign{1};
+  static constexpr float offset_rad_threshold{0.087f};  // (5 / 180) * M_PI
+  static constexpr float sign_threshold{0.1f};
 };
 
-template <class T>
-class DistortionCorrector : public DistortionCorrectorBase
+class DistortionCorrectorBase
 {
 protected:
   geometry_msgs::msg::TransformStamped::SharedPtr geometry_imu_to_base_link_ptr_;
   bool pointcloud_transform_needed_{false};
   bool pointcloud_transform_exists_{false};
   bool imu_transform_exists_{false};
-  rclcpp::Node * node_;
   std::unique_ptr<autoware::universe_utils::ManagedTransformBuffer> managed_tf_buffer_{nullptr};
 
   std::deque<geometry_msgs::msg::TwistStamped> twist_queue_;
   std::deque<geometry_msgs::msg::Vector3Stamped> angular_velocity_queue_;
 
-  void getIMUTransformation(const std::string & base_frame, const std::string & imu_frame);
-  void enqueueIMU(const sensor_msgs::msg::Imu::ConstSharedPtr imu_msg);
-  void getTwistAndIMUIterator(
+  rclcpp::Node * node_;
```

Comment:
can you make this a pointer const?
`rclcpp::Node * const node_;`

Reply:
@mojomex suggested using reference instead of the pointer in the concatenate PR but I didn't change it here.
Thanks for the catch, I changed it to reference in https://github.com/autowarefoundation/autoware.universe/pull/8380/commits/2594e761149e02d244614b41e8674b5e5864e833.

@knzo25 putting as and const reference will not work because the current design of managed_tf_buffer